catalyst-toolbox/src/rewards/dreps.rs

```diff
@@ -0,0 +1,238 @@
+        .ok_or(Error::Overflow)?;
+
+    let dreps = snapshot
+        .into_iter()
+        .filter(|v| v.hir.voting_group == drep_voting_group)
+        .collect::<Vec<_>>();
+
+    let total_dreps_stake = dreps
+        .iter()
+        .map(|d| u64::from(d.hir.voting_power))
+        .sum::<u64>();
+
+    let unique_dreps = dreps
+        .iter()
+        .map(|s| s.hir.voting_key.clone())
+        .collect::<HashSet<_>>();
+    if unique_dreps.len() != dreps.len() {
+        return Err(Error::MultipleEntries);
+    }
+
+    let filtered = filter_requirements(dreps, votes, top_dreps_to_reward, dreps_votes_threshold);
+
+    let res = filtered
+        .into_iter()
+        .map(|d| {
+            let reward = Decimal::from(u64::from(d.hir.voting_power))
+                / Decimal::from(total_active_stake)
+                * total_rewards;
+            (d.hir.voting_key, reward)
+        })
+        .collect::<BTreeMap<_, _>>();
+
+    let expected_rewards = if total_active_stake == 0 {
+        Decimal::ZERO
+    } else {
+        total_rewards * Decimal::from(total_dreps_stake) / Decimal::from(total_active_stake)
+    };
+    assert_are_close(res.values().sum(), expected_rewards);
+
+    Ok(res)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::snapshot::*;
+    use jormungandr_lib::crypto::hash::Hash;
+    use std::collections::HashMap;
+    use test_strategy::proptest;
+
+    #[proptest]
+    fn test_small(snapshot: Snapshot) {
+        let voting_keys = snapshot.voting_keys().collect::<Vec<_>>();
+
+        let votes_count = voting_keys
+            .iter()
+            .enumerate()
+            .map(|(i, &key)| {
+                (
+                    key.to_owned(),
+                    if i % 2 == 0 {
+                        HashSet::from([Hash::from([0u8; 32])])
+                    } else {
+                        HashSet::new()
+                    },
+                )
+            })
+            .collect::<VoteCount>();
+        let voters = snapshot.to_full_snapshot_info();
+        let voters_active = voters
+            .clone()
+            .into_iter()
+            .enumerate()
+            .filter(|(i, _utxo)| i % 2 == 0)
+            .map(|(_, utxo)| utxo)
+            .collect::<Vec<_>>();
+
+        let rewards = calc_dreps_rewards(
+            voters,
+            votes_count.clone(),
+            String::new(),
+            voting_keys.len(),
+            Threshold::new(1, HashMap::new(), Vec::new()).unwrap(),
+            Rewards::ONE,
+        )
+        .unwrap();
+        let rewards_no_inactive = calc_dreps_rewards(
+            voters_active,
+            votes_count.clone(),
+            String::new(),
+            voting_keys.len(),
+            Threshold::new(1, HashMap::new(), Vec::new()).unwrap(),
+            Rewards::ONE,
+        )
+        .unwrap();
+        // Rewards should ignore inactive voters
+        assert_eq!(rewards, rewards_no_inactive);
```

Comment:
Can we use `prop_assert_eq!()` here (and below)? It can significantly reduce the noise in the test output, which is helpful when tracking down the cause of a failing case?